python/lsst/sims/catUtils/utils/SNIaLightCurveGenerator.py

```diff
@@ -0,0 +1,164 @@
+                                        if obs.bandpass == bp_name])
+
+            m5_dict[bp_name] = np.array([obs.m5[bp_name] for obs in grp
+                                         if obs.bandpass == bp_name])
+
+            gamma_dict[bp_name] = np.array([calcGamma(self.lsstBandpassDict[bp_name],
+                                                      obs.m5[obs.bandpass], self.phot_params)
+                                            for obs in grp if obs.bandpass == bp_name])
+
+            if len(t_dict[bp_name]) > 0:
+                local_t_min = t_dict[bp_name].min()
+                local_t_max = t_dict[bp_name].max()
+                if t_min is None or local_t_min < t_min:
+                    t_min = local_t_min
+
+                if t_max is None or local_t_max > t_max:
+                    t_max = local_t_max
+
+        snobj = SNObject()
+
+        cat = cat_dict[cat_dict.keys()[0]]  # does not need to be associated with a bandpass
+
+        for chunk in query_result:
+            t_start_chunk = time.time()
+            for sn in cat.iter_catalog(query_cache=[chunk]):
+                sn_rng = self.sn_universe.getSN_rng(sn[1])
+                sn_t0 = self.sn_universe.drawFromT0Dist(sn_rng)
+                if sn[5] <= self.z_cutoff and np.isfinite(sn_t0) and \
+                    sn_t0 < t_max + cat.maxTimeSNVisible and \
+                    sn_t0 > t_min - cat.maxTimeSNVisible:
+
+                    sn_c = self.sn_universe.drawFromcDist(sn_rng)
+                    sn_x1 = self.sn_universe.drawFromx1Dist(sn_rng)
+                    sn_x0 = self.sn_universe.drawFromX0Dist(sn_rng, sn_x1, sn_c, sn[4])
+
+                    snobj.set(t0=sn_t0, c=sn_c, x1=sn_x1, x0=sn_x0, z=sn[5])
+
+                    for bp_name in cat_dict:
+                        t_list = t_dict[bp_name]
+                        m5_list = m5_dict[bp_name]
+                        gamma_list = gamma_dict[bp_name]
+                        bandpass = self.lsstBandpassDict[bp_name]
+                        if len(t_list) == 0:
+                            continue
+
+                        if snobj.maxtime() >= t_list[0] and snobj.mintime() <= t_list[-1]:
+                            active_dexes = np.where(np.logical_and(t_list >= snobj.mintime(),
+                                                                   t_list <= snobj.maxtime()))
+
+                            t_active = t_list[active_dexes]
+                            m5_active = m5_list[active_dexes]
+                            gamma_active = gamma_list[active_dexes]
+
+                            if len(t_active) > 0:
+                                wave_ang = bandpass.wavelen*10.0
+                                mask = np.logical_and(wave_ang > snobj.minwave(),
+                                                      wave_ang < snobj.maxwave())
+
+                                wave_ang = wave_ang[mask]
+                                snobj.set(mwebv=sn[6])
```

Comment:
'Why? The unittest validates the light curve generator against your InstanceCatalogs. They agree at the sub-millimag level.'

- Because having multiple methods to do the same thing in different parts of our codebase would make life really hard for usual reasons of avoiding code duplication. Essentially, I want all bug-fixes and improvements in extinction to be applied in the photUtils method, and all SN extinctions related issues to be in one place as far as possible.
- I know you caught the rectification issue, but I did not notice if you caught the case where the non-rectified SED has the outside the wavelength range problem.
- I would like to add a method to prevent accidental usage as I see this as something an user is likely to try (in analogy with other set methods) without noticing thee things like rectification etc. (which you did).